example/particle_tracers/particle_tracers.cpp

```diff
@@ -0,0 +1,515 @@
+//========================================================================================
+// Parthenon performance portable AMR framework
+// Copyright(C) 2021 The Parthenon collaboration
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+// (C) (or copyright) 2021. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include "particle_tracers.hpp"
+#include "Kokkos_CopyViews.hpp"
+#include "Kokkos_HostSpace.hpp"
+#include "basic_types.hpp"
+#include "config.hpp"
+#include "globals.hpp"
+#include "interface/metadata.hpp"
+#include "interface/update.hpp"
+#include "kokkos_abstraction.hpp"
+#include "parthenon/driver.hpp"
+#include "refinement/refinement.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#define SMALL (1.e-10)
+
+using namespace parthenon::driver::prelude;
+using namespace parthenon::Update;
+
+typedef Kokkos::Random_XorShift64_Pool<> RNGPool;
+
+namespace particles_example {
+
+Packages_t ProcessPackages(std::unique_ptr<ParameterInput> &pin) {
+  Packages_t packages;
+  packages.Add(particles_example::Particles::Initialize(pin.get()));
+  return packages;
+}
+
+namespace Particles {
+
+// *************************************************//
+// define the Particles package, including         *//
+// initialization and update functions.            *//
+// *************************************************//
+
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  auto pkg = std::make_shared<StateDescriptor>("particles_package");
+
+  Real vmatx = pin->GetOrAddReal("Material", "vx", 1.0);
```

Comment:
I don't love the name `Material` here. What about `background`?

Reply:
Yeah that's good, switched to `Background`